pkgs/applications/editors/pulsar/default.nix

```diff
@@ -0,0 +1,159 @@
+
+      # Remove lines 114 to 116 (inclusive) and replace with ''${localePaths}
+      sed -i '114,116d' ./app/node_modules/spell-check/lib/locale-checker.js
+      echo 'sed -i "114i ${localePatchs}" ./app/node_modules/spell-check/lib/locale-checker.js'
+      sed -i "114i ${localePatchs}" ./app/node_modules/spell-check/lib/locale-checker.js
+      # Rebuild app.asar and clean up
+      asar pack ./app $opt/resources/app.asar
+      rm -rf ./app
+    fi
+
+    gappsWrapperArgs+=(
+      # needed for gio executable to be able to delete files
+      --prefix "PATH" : "${glib.bin}/bin"
+    )
+  '';
+
+  postFixup = ''
+    opt=$out/opt/Pulsar
+    # Patch the prebuilt binaries
+    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      --set-rpath "${newLibpath}:$opt" \
+      --add-needed libffmpeg.so \
+      --add-needed libxshmfence.so.1 \
+      --add-needed libxkbcommon.so.0 \
+      --add-needed libxkbfile.so.1 \
+      --add-needed libsecret-1.so.0 \
+      $opt/pulsar
+    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      --set-rpath "${newLibpath}" \
+      $opt/resources/app/ppm/bin/node
+    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      $opt/resources/app.asar.unpacked/node_modules/symbols-view/vendor/ctags-linux
+
+    # Replace the bundled git with the one from nixpkgs
+    dugite=$opt/resources/app.asar.unpacked/node_modules/dugite
+    rm -f $dugite/git/bin/git
+    ln -s ${pkgs.git}/bin/git $dugite/git/bin/git
+    rm -f $dugite/git/libexec/git-core/git
+    ln -s ${pkgs.git}/bin/git $dugite/git/libexec/git-core/git
+
+    # Patch the bundled node executables
+    find $opt -name "*.node" -exec patchelf --set-rpath "${newLibpath}:$opt" {} \;
+
+    # Create a wrapper script for the executable
+    mkdir -p $out/bin
+    cat > $out/bin/${name} <<EOF
+    #!${stdenv.shell}
+    nohup $opt/pulsar --no-sandbox "\$@" > /dev/null 2>&1 &
+    EOF
+    chmod +x $out/bin/${name}
+
+    # Create a desktop file
+    mkdir -p $out/share/applications
+    cat > $out/share/applications/${name}.desktop <<EOF
+    [Desktop Entry]
+    Name=Pulsar
+    Comment=A Community-led Hyper-Hackable Text Editor
+    Exec=$out/bin/${name}
+    Icon=$out/opt/Pulsar/resources/pulsar.png
+    Terminal=false
+    Type=Application
+    Categories=Development;TextEditor;
+    MimeType=text/plain;
+    EOF
+  '';
+
+  meta = with lib; {
+    description = "A Community-led Hyper-Hackable Text Editor";
+    longDescription = ''
+      A Community-led Hyper-Hackable Text Editor, Forked from Atom, built on Electron.
+      Designed to be deeply customizable, but still approachable using the default configuration.
+    '';
+    homepage = "https://github.com/pulsar-edit/pulsar";
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
+    license = licenses.mit;
+    platforms = platforms.x86_64;
+    broken = stdenv.targetPlatform.system != "x86_64-linux";
```

Comment:
If we add `aarch64` to platforms we should also allow that here..
Though, broken is not quite right anyway. Isn't there more precise platforms..?
platforms should just be `x86_64-linux` and `aarch64-linux` unless you want to figure out darwin..

Reply:
https://nixos.org/manual/nixpkgs/stable/#sec-allow-unsupported-system

> The difference between a package being unsupported on some system and being broken is admittedly a bit fuzzy. If a program ought to work on a certain platform, but doesn’t, the platform should be included in `meta.platforms`, but marked as broken with e.g. `meta.broken = !hostPlatform.isWindows`. Of course, this begs the question of what "ought" means exactly. That is left to the package maintainer.

My interpretation: "ought to work on x" is what the software should run on, so I would include Aarch64 and Darwin in `platforms`. But since the packaging only fetches & handles the archive for `x86_64-linux` right now, I would add everything else to `broken`. Once `aarch64-linux` or `*-darwin` is actually handled, I would then adjust `broken` accordingly.

In the end, the only difference is whether the error message says "not supported" or "broken". My suggestions for both `platforms` and `broken` are how I would handle them, ultimately I'm fine with leaving them as is if whoever maintains this wants to have them like that though.

Reply:
The issue is not strictly that I don't want to support aarch64 (actually, there is a Apple package so the package could target Darwin too), it's just that I do not own any non x86-64 Linux systems, so I have no way to decently test if the package is working